### ISSUE TITLE
pachctl copy-file does nothing

### ISSUE THREAD

**Issue by mdaniel** [pachyderm/pachyderm#3449]
**What happened?**:

Invoking `copy-file` from one repo to another did nothing, but did not emit any error message

```console
$ pachctl list-repo
NAME                CREATED      SIZE
ingress-batch-files 3 hours ago  0B
atp                 21 hours ago 22.95KiB

$ pachctl list-file ingress-batch-files master
COMMIT                           NAME     TYPE COMMITTED   SIZE
e1b9d85b3bf042b5a79efb5420c95c0e /batches dir  3 hours ago 1010MiB

$ pachctl list-file atp master
COMMIT                           NAME   TYPE COMMITTED     SIZE
f5007fa868814a50a55592454bb2ee90 /_meta dir  8 minutes ago 11.48KiB

$ pachctl -vvv copy-file ingress-batch-files master /batches atp master /batches/001_1
[etcd/grpc] INFO  2019/02/05 18:36:23 log.go:172: INFO: 2019/02/05 18:36:23 parsed scheme: ""
[etcd/grpc] INFO  2019/02/05 18:36:23 log.go:172: INFO: 2019/02/05 18:36:23 scheme "" not registered, fallback to default scheme
[etcd/grpc] INFO  2019/02/05 18:36:23 log.go:172: INFO: 2019/02/05 18:36:23 ccResolverWrapper: sending new addresses to cc: [{pach.example.com:650 0  <nil>}]
[etcd/grpc] INFO  2019/02/05 18:36:23 log.go:172: INFO: 2019/02/05 18:36:23 ClientConn switching balancer to "pick_first"
[etcd/grpc] INFO  2019/02/05 18:36:23 log.go:172: INFO: 2019/02/05 18:36:23 pickfirstBalancer: HandleSubConnStateChange: 0xc420547e80, CONNECTING
[etcd/grpc] INFO  2019/02/05 18:36:23 log.go:172: INFO: 2019/02/05 18:36:23 pickfirstBalancer: HandleSubConnStateChange: 0xc420547e80, READY

$ echo $?
0

$ pachctl list-commit atp
REPO COMMIT                           PARENT                           STARTED        DURATION           SIZE
atp  b99c107dc27145b1a018d29b4a341fc0 f5007fa868814a50a55592454bb2ee90 37 seconds ago -                  -
$ pachctl finish-commit atp b99c107dc27145b1a018d29b4a341fc0
$ pachctl list-commit atp
REPO COMMIT                           PARENT                           STARTED        DURATION           SIZE
atp  b99c107dc27145b1a018d29b4a341fc0 f5007fa868814a50a55592454bb2ee90 51 seconds ago 49 seconds         11.48KiB

$ pachctl list-file atp master
COMMIT                           NAME   TYPE COMMITTED      SIZE
b99c107dc27145b1a018d29b4a341fc0 /_meta dir  18 seconds ago 11.48KiB
```

_p.s. I also tried the following, with the exact same result_
```sh
into_commit=`pachctl start-commit atp master`
for i in `pachctl list-file ingress-batch-files master /batches --raw | jq -r .file.path`
do
    bn=`basename $i`
    pachctl -vv copy-file ingress-batch-files master $i atp $into_commit /batches/001_1/$bn || break
done
pachctl finish-commit atp $into_commit
```

**What you expected to happen?**:

Copy the file or say why not

**How to reproduce it (as minimally and precisely as possible)?**:

**Anything else we need to know?**:

**Environment?**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.2", GitCommit:"cff46ab41ff0bb44d8584413b598ad8360ec1def", GitTreeState:"clean", BuildDate:"2019-01-15T02:23:27Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:00:57Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```
- Pachyderm CLI version (use `pachctl version`):
```
COMPONENT           VERSION
pachctl             1.8.2
pachd               1.8.2
```
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s): `aws`
- OS (e.g. from /etc/os-release):
- Others:

Here are the (nicely formatted) log lines from the only `pachd` Pod showing some relevant lines -- there are no `ERROR` lines

```
2019-02-06T02:11:48Z INFO pfs.API.CopyFile 
{
  "request": {
    "src": {
      "commit": {
        "repo": {
          "name": "ingress-batch-files"
        },
        "id": "master"
      },
      "path": "/batches/EXAMPLE_EXAMPLE.cv_time.0.gz"
    },
    "dst": {
      "commit": {
        "repo": {
          "name": "atp"
        },
        "id": "master"
      },
      "path": "/batches/001_1/EXAMPLE_EXAMPLE.cv_time.0.gz"
    }
  }
}
2019-02-06T02:11:48Z INFO authclient.API.WhoAmI {"request":{}} []
2019-02-06T02:11:48Z INFO authclient.API.WhoAmI {"request":{}} []
2019-02-06T02:11:48Z INFO authclient.API.WhoAmI {"request":{}} []
2019-02-06T02:11:48Z INFO authclient.API.WhoAmI {"request":{}} []
2019-02-06T02:11:48Z INFO authclient.API.WhoAmI {"request":{}} []
2019-02-06T02:11:48Z INFO pfs.API.CopyFile 
{
  "duration": 0.016745363,
  "request": {
    "src": {
      "commit": {
        "repo": {
          "name": "ingress-batch-files"
        },
        "id": "e1b9d85b3bf042b5a79efb5420c95c0e"
      },
      "path": "/batches/EXAMPLE_EXAMPLE.cv_time.0.gz"
    },
    "dst": {
      "commit": {
        "repo": {
          "name": "atp"
        },
        "id": "509a9200b6c4498d80b9e01e857cd502"
      },
      "path": "/batches/001_1/EXAMPLE_EXAMPLE.cv_time.0.gz"
    }
  },
  "response": {}
}
2019-02-06T02:11:48Z INFO pfs.API.CopyFile 
{
  "request": {
    "src": {
      "commit": {
        "repo": {
          "name": "ingress-batch-files"
        },
        "id": "master"
      },
      "path": "/batches/EXAMPLE_EXAMPLE.cv_time.1.gz"
    },
    "dst": {
      "commit": {
        "repo": {
          "name": "atp"
        },
        "id": "master"
      },
      "path": "/batches/001_1/EXAMPLE_EXAMPLE.cv_time.1.gz"
    }
  }
}
2019-02-06T02:11:48Z INFO authclient.API.WhoAmI {"request":{}} []
2019-02-06T02:11:48Z INFO authclient.API.WhoAmI {"request":{}} []
2019-02-06T02:11:48Z INFO authclient.API.WhoAmI {"request":{}} []
2019-02-06T02:11:48Z INFO authclient.API.WhoAmI {"request":{}} []
2019-02-06T02:11:48Z INFO authclient.API.WhoAmI {"request":{}} []
2019-02-06T02:11:48Z INFO pfs.API.CopyFile 
{
  "duration": 0.017641499,
  "request": {
    "src": {
      "commit": {
        "repo": {
          "name": "ingress-batch-files"
        },
        "id": "e1b9d85b3bf042b5a79efb5420c95c0e"
      },
      "path": "/batches/EXAMPLE_EXAMPLE.cv_time.1.gz"
    },
    "dst": {
      "commit": {
        "repo": {
          "name": "atp"
        },
        "id": "509a9200b6c4498d80b9e01e857cd502"
      },
      "path": "/batches/001_1/EXAMPLE_EXAMPLE.cv_time.1.gz"
    }
  },
  "response": {}
}
```

**Comment by gabrielgrant** [comment]
@mdaniel to confirm, are these both input repos, or are you copying an output?

**Comment by mdaniel** [comment]
It's a copy from an output back to an input(? I guess that's the terminology for a "non-output" repo?); the `ingress-batch-files` in the above output is a repo that was created by a pipeline job, and the `atp` in the above output was a "root" repo, created by `create-repo`

IIRC Joey said in Slack that it is unsupported behavior to do that, but if that's true, then having `pachctl` exit non-zero and/or be chatty about the situation would be far, far better than what appeared to be a no-op

**Comment by gabrielgrant** [comment]
Yes, apparently the recent performance improvements to output repos have made it so files can't be copied out of them to input repos. Agree that either these copies should be allowed, or they should fail in a more obvious manner

**Comment by da-i** [comment]
Just ran into this issue as well,
`$ pachctl list branch rdl-final
BRANCH  HEAD
staging 395b80de7d3a410eb39e241d6caea096
master  6bedb562fa9d4db58b616193b61ceb76
$ pachctl list file rdl-final@staging
NAME TYPE SIZE
$ pachctl copy file rdl-final@master:/SH01/SH01.normal.bam rdl-final@staging:/SH01/SH01.normal.bam -v
[0000]  INFO parsed scheme: "" source=etcd/grpc
[0000]  INFO scheme "" not registered, fallback to default scheme source=etcd/grpc
[0000]  INFO ccResolverWrapper: sending update to cc: {[{10.142.15.229:30650 0  <nil>}] } source=etcd/grpc
[0000]  INFO ClientConn switching balancer to "pick_first" source=etcd/grpc
[0000]  INFO pickfirstBalancer: HandleSubConnStateChange: 0xc0003454a0, CONNECTING source=etcd/grpc
[0000]  INFO pickfirstBalancer: HandleSubConnStateChange: 0xc0003454a0, READY source=etcd/grpc
$ pachctl list file rdl-final@staging
NAME TYPE SIZE`

**Comment by jdoliner** [comment]
Fixed by #4475, this will ship in 1.10.0.
